### ISSUE TITLE
(gdk): gdk creates bucket even if the component does not have any artifact

### ISSUE THREAD

**Issue by massi-ang** [aws-greengrass/aws-greengrass-gdk-cli#87]
**Describe the bug**
 gdk creates bucket even if the component does not have any artifact

**To Reproduce**
create a recipe that does not have any artifacts and do `gdk component publish`

**Expected behavior**
Bucket property should not be required and bucket should not be created as there are no artifacts

**Actual behavior**
bucket is requires, and bucker is created, requiring also permissions to do so

**Environment**
 - OS: [e.g. Ubuntu 20.04] (If possible, provide output of `uname -a`)
 - JDK version:
 - Nucleus version:

**Additional context**

E.g. what is the impact of the bug?

**Comment by saranyailla** [comment]
Hi Massi, 
Thanks for the feedback, Massi. This issue is fixed as part of the PR  https://github.com/aws-greengrass/aws-greengrass-gdk-cli/pull/58 and is released with gdk v1.1.0. 

Please install the latest cli version using v1.1.0 tag as shown below and let us know if you've more questions. 
`pip3 install git+https://github.com/aws-greengrass/aws-greengrass-gdk-cli.git@v1.1.0`

**Comment by massi-ang** [comment]
Thanks for the update, but it seems that "bucket" is still required, and in the logs you get some info about the tool uploading data to such bucket:

```
[2022-03-02 11:08:04] INFO - Uploading the component built artifacts to s3 bucket.
[2022-03-02 11:08:04] INFO - Uploading component artifacts to S3 bucket: <placeholder>-eu-west-1-699391019698. If this is your first time using this bucket, add the 's3:GetObject' permission to each core device's token exchange role to allow it to download the component artifacts. For more information, see https://docs.aws.amazon.com/greengrass/v2/developerguide/device-service-role.html.
```
